src/clients/c++/request.h

```diff
@@ -666,6 +667,29 @@ class InferContext {
   Error GetReadyAsyncRequest(
       std::shared_ptr<Request>* async_request, bool wait);
 
+  /// Send current request through the context's stream. All requests sent will
+  /// be directed to the same model instance, be processed, and be returned
+  /// in the same order. Note that you need to retrieve the responses with
```

Comment:
Does GRPC stream guarantee that responses are in same order as requests?

Reply:
Not exactly, gRPC guarantees that the receiving order will be the same as the order that the other side sends back. So here is based on the assumption that the server will process the requests in sequence.

Reply:
I general that is not true... when you call into HandleInfer there is no guarentee that the completion callbacks will be called in that same order. You will need to add that ordering to your server-side implementation. Is seems like the stream / non-stream aspect needs to be part of the InferContext. You can still send sync or async through that context and it will just send those requests over a stream or not.

Reply:
Right now in the server-side streaming, the requests will be queued up and only be processed if the previous request is finished in the stream. So if the client is not using correlation ID, then using the streaming API is not desired.

I think stream / non-stream aspect sounds right. And I think, if the difference between sync/async infer is whether multiple requests can be initiated at the same time, then the difference between sync/async streaming is in terms of streams, whether multiple streams can be maintained in the InferContext at the same time.